src/Illuminate/Http/Client/Factory.php

```diff
@@ -267,9 +267,19 @@ public function stubUrl($url, $callback)
                 return;
             }
 
-            return $callback instanceof Closure || $callback instanceof ResponseSequence
-                        ? $callback($request, $options)
-                        : $callback;
+            if (is_int($callback) && $callback >= 100 && $callback < 600) {
+                return static::response(status: $callback);
```

Comment:
Wow. I took another look at the code and was very surprised!

@jasonmccreary, okay, let's say `200` is the status code and `204` is the response body.

But what is the difference between, for example, `500` and `600`? Here `500` is the **status code**, and `600` is the **response body** 😀

@taylorotwell, you'll be interested too (I wrote my thoughts above, and here's an addendum).

```php
Http::fake([
    'google.com' => 'Hello World', // no problem
    'github.com' => ['foo' => 'bar'], // no problem

    'bar.laravel.com' => '204', // it's a response body with status code 200

    'foo.laravel.com' => 99, // it's a response body with status code 200
    'foo.laravel.com' => 100, // it's a status code 100
    'foo.laravel.com' => 204, // it's a status code 204
    'foo.laravel.com' => 599, // it's a status code 599
    'foo.laravel.com' => 600, // it's a response body with status code 200
]);
```

Reply:
My suggestion:

```php
public function stubUrl($url, $callback)
{
    return $this->fake(function ($request, $options) use ($url, $callback) {
        if (! Str::is(Str::start($url, '*'), $request->url())) {
            return;
        }

        if ($callback instanceof Closure || $callback instanceof ResponseSequence) {
            return $callback($request, $options);
        }

        if (is_numeric($callback) || is_string($callback) || is_array($callback)) { // here
            return static::response($callback);
        }

        return $callback;
    });
}
```

Reply:
What about just wrapping the callback? This would be following the original train of thought of this PR:
```diff
public function stubUrl($url, $callback)
{
    return $this->fake(function ($request, $options) use ($url, $callback) {
        if (! Str::is(Str::start($url, '*'), $request->url())) {
            return;
        }

+        if (is_numeric($callback) || is_string($callback) || is_array($callback) {
+            $callback = static::response($callback);
+        }

        if ($callback instanceof Closure || $callback instanceof ResponseSequence) {
            return $callback($request, $options);
        }
-
-        if (is_numeric($callback) || is_string($callback) || is_array($callback)) { // here
-            return static::response($callback);
-        }

        return $callback;
    });
}
```

Reply:
Yeah, you can do that too :)

Reply:
Oh, I misread `ResponseSequence` as `Response` 🤦🏻‍♂️ Your suggestion makes more sense then 😅